### PR TITLE
build, windows: do not fail on Windows SDK Prompt

### PR DESCRIPTION
**Pull request by schlamar**

### CODE REVIEW

**Comment by schlamar** [comment]
Hello bot, I have already signed the CLA :-)

File: vcbuild.bat [hunk 1]
```diff
@@ -111,8 +111,9 @@ echo Project files generated.
 if defined nobuild goto run
 
 @rem Check if VS build env is available
-if not defined VCINSTALLDIR goto msbuild-not-found
-goto msbuild-found
+if defined VCINSTALLDIR goto msbuild-found
+if defined WindowsSDKDir goto msbuild-found
+goto msbuild-not-found
```
Comment (saghul) [inline vcbuild.bat]: This label seems useless now since we only jump from the line above, right?

**Comment by saghul** [comment]
I don't know what that "Windows SDK" is, does this LGTY @piscisaureus?

File: vcbuild.bat (hunk 1 above)
Comment (schlamar) [inline vcbuild.bat]: Ah right, will fix that.

**Comment by schlamar** [comment]
> I don't know what that "Windows SDK" is, does this LGTY @piscisaureus?

This is a free tool from Microsoft which allows to compile Visual Studio projects on x64 (
http://www.microsoft.com/en-us/download/details.aspx?id=8279). So this is an alternative to VS Express as build tool because VS Express can only compile x86.

**Comment by saghul** [comment]
Hey @schlamar, I installed the Windows SDK and I get the following error when running `vcbuild.bat test`:

```
  runner-win.c
LINK : fatal error LNK1123: failure during conversion to COFF: file invalid or
corrupt [c:\Users\saghul\src\libuv\run-tests.vcxproj]
```

I have Visual Studio Express 2008, 2013 and the Windows SDK 7.1 installed. Any ideas?

**Comment by schlamar** [comment]
Are you in the Windows SDK prompt and are you targeting x86 debug (`setenv /x86 /debug`)?

Works for me with VS 2008, VS Express 2010 and Windows SDK 7.1 (except the test failures noted in #711).

**Comment by schlamar** [comment]
Maybe you need to `set GYP_MSVS_VERSION=2010`. AFAIS Windows SDK 7.1 is targeting VS 2010 projects.

**Comment by saghul** [comment]
I don't have VS 2010 :-/ I guess vcbuild.bat is detecting my 2013 version, should that work? Else, is there a way to for my 2008 express version to be forced?

**Comment by schlamar** [comment]
> I don't have VS 2010

This should work anyway. I can create a VS 2013 project with `set GYP_MSVS_VERSION=2013` without having it installed. So with `GYP_MSVS_VERSION=2010` it should create a VS 2010 project which then could be compiled with the SDK (without having VS 2010 installed).

For 2008 you can either `set GYP_MSVS_VERSION=2008` and run it in the VS 2008 command prompt or prevent detection of 2013 by `set VS120COMNTOOLS=`.

**Comment by saghul** [comment]
I tried setting it to use VS 2013 extress and this is what I get:

```
c:\Users\saghul\src\libuv>set GYP_MSVS_VERSION=2013e

c:\Users\saghul\src\libuv>vcbuild.bat test
['-Dtarget_arch=ia32', '-Duv_library=static_library', '.\\uv.gyp', '-I', '.\\com
mon.gypi', '--depth=.', '-Dhost_arch=x86', '-Dcomponent=static_library']
Project files generated.
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
c:\Users\saghul\src\libuv\libuv.vcxproj(18,3): error MSB4019: The imported proj
ect "C:\Program Files\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cpp.Default.pro
ps" was not found. Confirm that the path in the <Import> declaration is correct
, and that the file exists on disk.
c:\Users\saghul\src\libuv\uv_dtrace_header.vcxproj(18,3): error MSB4019: The im
ported project "C:\Program Files\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cpp.
Default.props" was not found. Confirm that the path in the <Import> declaration
 is correct, and that the file exists on disk.
c:\Users\saghul\src\libuv\uv_dtrace_provider.vcxproj(18,3): error MSB4019: The
imported project "C:\Program Files\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cp
p.Default.props" was not found. Confirm that the path in the <Import> declarati
on is correct, and that the file exists on disk.

c:\Users\saghul\src\libuv>
```

Of course, I don't have that file. Should I install something else?

**Comment by schlamar** [comment]
The error with 2013 should be not relevant to my patch. Might be an issue with gyp.

Looks like the linker error is a MS bug which can be resolved with a patch: http://www.microsoft.com/en-us/download/details.aspx?id=4422. You should be able to install the patch even if you have not installed VS 2010 (see http://www.chromium.org/developers/how-tos/build-instructions-windows#TOC-Setting-up-the-environment-for-building-with-Visual-C-2010-Express-or-Windows-7.1-SDK).

**Comment by schlamar** [comment]
Rebased against master. 

BTW, I have seen the linker error on x64 test build, too. But it was successfully resolved with the patch from MS linked above.

**Comment by saghul** [comment]
Thanks for the help! I'll try that later and let you know how it goes!

**Comment by saghul** [comment]
Success! That patch made it! I'll land it in a bit.

**Comment by saghul** [comment]
Thanks Marc, landed in https://github.com/joyent/libuv/commit/2f6d4b4b58ea290961dc983117d31b19e198403a

**Comment by schlamar** [comment]
@saghul and this would be also required in v0.10 for proper pyuv builds.

**Comment by saghul** [comment]
Ok, I'll backport them both. Thanks for letting us know!

**Comment by saghul** [comment]
Backport landed in https://github.com/joyent/libuv/commit/cd6db8bbefb11a780977052f1a561920d6d60bbe

**Pull request closed without merging**